### PR TITLE
Allows the sharing modal to search on it's own (BREAKING)

### PR DESCRIPTION
**Pull request by ryan-bradford**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [ ] Examples have been added / updated (for bug fixes / features)
-   [x] Changelog has been updated

## PR Type

What kind of change does this PR introduce?

-   [x] Bugfix
-   [x] Feature
-   [ ] Code style update (formatting, local variables)
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Build related changes
-   [ ] CI related changes
-   [ ] Documentation content changes
-   [ ] Example website changes
-   [ ] Version bump
-   [ ] Other... Please describe:

## What does this change do?

Allows the users, groups, orgs sharing modal directive to supply default values for both makeSearch and entityRenderer. Also, fixes two bugs in the sharing modal: error on loading with the directive, too many search warnings being shown. See below for GIF.

## What manual testing did you do?

Loaded into VCD_UI and created a modal with the users groups orgs directive. Was able to search for entities and add to the datagrid. 

## Screenshots (if applicable)

users-groups-orgs

## Does this PR introduce a breaking change?

-   [X] Yes
-   [ ] No

# Important!!

Adds the Angular Client as a dependency to this project. This forces clients of our components to both install the angular client, but also setup SystemJs to provide @vcd/common. This is done automatically, but is not done automatically for any component used outside VCD_UI.

Additionally, using href as the ID was a poor choice in the first place. This commit changes it to use ID as the ID property instead of HREF. This is a breaking change.

### CODE REVIEW

**Comment by juanmendes** [comment]
Can we add a separate example of this within our examples container?

**Comment by ryan-bradford** [comment]
> Can we add a separate example of this within our examples container?

We cannot yet add a separate example for this within the examples container. This requires us to build a framework to log into a TestBed. However, I'm going to add the following to the review:
1. Better text on each of the examples.
2. A more clear first example in how it's different than the second
3. Clicking the label toggles the checkbox.

File: projects/components/src/sharing-modal/sharing-modal.component.ts [hunk 1]
```diff
@@ -170,6 +170,8 @@ export class SharingModalComponent implements OnInit {
     }
 
     ngOnInit(): void {
-        this.tabs = [...this.tabs];
+        if (this.tabs) {
+            this.tabs = [...this.tabs];
```
[the remark this answers is not in the document]
Comment (ryan-bradford) [reply projects/components/src/sharing-modal/sharing-modal.component.ts:174]: There was a problem where tabs was initially undefined in some cases. Changed it to initialize to an empty list.

File: projects/components/src/sharing-modal/users-groups-orgs-sharing-modal.directive.ts [hunk 2]
```diff
@@ -42,6 +48,8 @@ export class UsersGroupsOrgsSharingModalDirective implements OnInit {
             id: 'groups',
             title: this.translationService.translateAsync('vcd.cc.sharing.groups'),
             selectAllText: this.translationService.translateAsync('vcd.cc.sharing.groups.selected'),
+            makeSearch: this.makeSearchGenerator<GroupType>('group'),
```
[the remark this answers is not in the document]
Comment (ryan-bradford) [reply projects/components/src/sharing-modal/users-groups-orgs-sharing-modal.directive.ts]: Created a smarter type system. This should be fixed

File: projects/components/src/test.ts [hunk 3]
```diff
@@ -4,8 +4,11 @@
  */
 
 // This file is required by karma.conf.js and loads recursively all the .spec and framework files
+/** Patches in SystemJs for use by the Angular Client */
```
[the remark this answers is not in the document]
Comment (ryan-bradford) [reply projects/components/src/test.ts:7]: I updated this comment, but I did not file an issue. I don't think they need to change their library for our use case (installing and optionally using their services).

**Review by juanmendes — APPROVED** [review]

**Pull request merged**